index.js

```diff
@@ -21,14 +21,17 @@ var toArray = Array.from || (function (slice) {
 
 // ===================================================================
 
-function eventToPromise (emitter, event, opts) {
-  var ignoreErrors = opts && opts.ignoreErrors
+function eventToPromise (emitter, event, _opts) {
+  var opts = _opts || {}
+  var ignoreErrors = opts.ignoreErrors
+  var errorEvent = opts.error || 'error'
 
   return new Bluebird(function (resolve, reject) {
     var addListener =
+      emitter.once ||
+      emitter.on ||
```

Comment:
Some implementations (IIRC, socket.io) does not expose `once()` and [DOM event emitters](http://devdocs.io/dom/eventtarget) in browsers only have `addEventListener()` and `removeEventListener()`.